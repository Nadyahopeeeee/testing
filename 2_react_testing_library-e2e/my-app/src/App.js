// import { useState, useEffect } from 'react';

// import logo from './logo.svg';
// import './App.css';

// import Users from './users/Users';

// function App() {
//   const [data, setData] = useState(null)
//   const [toggle, setToggle] = useState(false)
//   const [value, setValue] = useState('')

//   const onClick = () => setToggle(prev => !prev)

//   useEffect(() => {
//     setTimeout(() => {
//       setData({})
//     }, 1000)
//   }, [])


//   return (
//     <div className="App">
//       {toggle === true && <div data-testid='toggle-elem'>toggle</div>}
//       {data && <div style={{color: 'red'}}>data</div>}
//       <h1>Hello world</h1>
//       <button data-testid='toggle-btn' onClick={onClick}>click me</button>
//       <input onChange={e => setValue(e.target.value)} type='text' placeholder='input value...'/>
//       <h2 data-testid='value-elem'>{value}</h2>
//       <Users/>
//     </div>
//   );
// }

// export default App;
