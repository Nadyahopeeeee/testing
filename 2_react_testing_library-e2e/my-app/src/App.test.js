import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';


describe('TEST APP', () => {
  test('render start layout', () => {
    render(<App />);
    const helloElement = screen.getByText(/hello world/i);
    const btn = screen.getByRole('button');
    const input = screen.getByPlaceholderText(/input value/i);
    expect(helloElement).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test('render data', async () => {
    render(<App />);
    screen.debug();
    const dataEl = await screen.findByText(/data/i)
    expect(dataEl).toBeInTheDocument();
    expect(dataEl).toHaveStyle({color: 'red'});
    screen.debug()

  });

  test('click event', () => {
    render(<App />);
    const btn = screen.getByTestId('toggle-btn')

    expect(screen.queryByTestId('toggle-elem')).toBeNull()
    fireEvent.click(btn)
    expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument();
    fireEvent.click(btn)
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
  });

  test("INPUT EVENT", () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/input value/i)
    expect(screen.queryByTestId('value-elem')).toContainHTML('');
    // Искуственное событие
    fireEvent.input(input, {
      target: {value: '123123'}
    })
    // // Близко к пользователю, обрабатываются события нажатия клавиш и тд
    // userEvent.type(input, '123123')
    expect(screen.queryByTestId('value-elem')).toContainHTML('123123');
  })
})

