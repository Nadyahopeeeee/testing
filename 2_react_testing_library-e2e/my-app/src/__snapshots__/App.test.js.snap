// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP render start layout 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
